### DISCUSSION TITLE
What is the update frequency that will not get you banned?

### DISCUSSION THREAD

**Discussion by xsherlockpl** [bimmerconnected/bimmer_connected#686] [Q&A]
Hi ALL,

MyBMWAuthError: HTTPStatusError: Client error '429 Too Many Requests' for url 'https://customer.bmwgroup.com/gcdm/oauth/authenticate'

So after the latest captcha update I  noticed that even every 30 min is too much for Euro server.
What are frequencies that work for you guys? 
What is the cooldown when you exceed and get banned?

**Comment by rikroe** [comment]
Are you using the CLI? Or have it integrated in HA?

My HA is running fine with the 5 minute polling interval (just created a new refresh token and injected it).

**Comment by xsherlockpl** [comment]
As I use OpenHab so.
I use CLI script that runs with cron. then convert results to mqtt with small python script.
It used to work with 15 min interval for me for first 2-3 years then 30min ... but now I get too many requests,  will let it cooldown for a day and start with  1h again 
Are you europe or US?

**Reply by rikroe** [reply]
Europe. But I have the feeling that CLI somehow reacts different than HA.
However there seems to be be some issue (see #681).

Will check, but earliest next week.

**Reply by rikroe** [reply]
https://github.com/bimmerconnected/bimmer_connected/pull/690 should hopefully help your login/captcha issues once released.

**Comment by xsherlockpl** [comment]
My problem was that for last 2 days I could not even login  on the web to my account.  "Technical problem, try again later" Now I got unbanned and I can run the script  with captcha-token from CLI multiple times in row with the same captcha (tried 3-4 as I dont want to get blocked again" . let see how long it will work

**Comment by ElZorroofOz** [comment]
Same as you polling from CLI (not cron, a shell loop but same), 15min works fine for a while generally.
I get auth errors but I thnk they may be a different problem (see other Q&A #698)

**Reply by xsherlockpl** [reply]
after 2 days I could finally login over web to my bmw account, and with 17.2 deployed I'm back to 30min updates and it is working.
That cat and mouse play with  BMW is tiring.   Their kuntish  IT policy is really determinig my next car purchase and I wish they understood that.

I need my car SOC to to efectively manage my home battery and car battery, So I buy a car that has a decent api for that.
